hosting/test_auth.py

```diff
@@ -0,0 +1,409 @@
+                    'hosting:validate',
+                    kwargs={
+                        'validate_slug': new_user.validation_slug
+                    }
+                ),
+                'dcl_text': settings.DCL_TEXT
+            },
+            'template_name': template_name,
+            'template_path': template_path
+        }
+        template_full_path = '%s%s' % (template_path, template_name)
+        text_content = render_to_string('%s.txt' % template_full_path, email_data['context'])
+        html_content = render_to_string('%s.html' % template_full_path, email_data['context'])
+        email = EmailMultiAlternatives(email_data['subject'], text_content)
+        email.attach_alternative(html_content, "text/html")
+        email.to = [email_data.get('to', 'info@ungleich.ch')]
+        email.send()
+        self.assertEqual(mail.outbox[1].body, text_content)
+        self.assertEqual(mail.outbox[1].alternatives[0][0], html_content)
+
+    def test_mail_type_de_content_to_user(self):
+        activate('de')
+        request = self.factory.get(reverse('hosting:signup'))
+        base_url = "{0}://{1}".format(request.scheme, request.get_host())
+        new_user = CustomUser.objects.get(email='test@example.com')
+        template_path = 'datacenterlight/emails/'
+        template_name = 'user_activation'
+        email_data = {
+            'subject': '{dcl_text} {account_activation}'.format(
+                dcl_text=settings.DCL_TEXT,
+                account_activation=_('Account Activation')
+            ),
+            'from_address': settings.DCL_SUPPORT_FROM_ADDRESS,
+            'to': new_user.email,
+            'context': {
+                'base_url': base_url,
+                'activation_link': reverse(
+                    'hosting:validate',
+                    kwargs={
+                        'validate_slug': new_user.validation_slug
+                    }
+                ),
+                'dcl_text': settings.DCL_TEXT
+            },
+            'template_name': template_name,
+            'template_path': template_path
+        }
+        template_full_path = '%s%s' % (template_path, template_name)
+        text_content = render_to_string('%s.txt' % template_full_path, email_data['context'])
+        html_content = render_to_string('%s.html' % template_full_path, email_data['context'])
+        email = EmailMultiAlternatives(email_data['subject'], text_content)
+        email.attach_alternative(html_content, "text/html")
+        email.to = [email_data.get('to', 'info@ungleich.ch')]
+        email.send()
+        self.assertEqual(mail.outbox[1].body, text_content)
+        self.assertEqual(mail.outbox[1].alternatives[0][0], html_content)
+        email_text = email.alternatives[0][0]
+        if "You can activate your Data Center Light account" in email_text:
+            language = 'en-us'
+        else:
+            language = "de"
+        self.assertEqual(language, 'en-us')
```

Comment:
Isn't this a wrong check ? We have activated EN and we are asserting the language to EN.

Reply:
When I write code as self.assertEqual(language, 'de'), test gave me error.
Because the email translation with 'DE' functionality wasn't implemented to this project at this moment.
The correct unit test script should be self.assertEqual(language, 'de').
But it occurs error on Travis CI.
So I changed code as above.

Reply:
So, something's not good ..